pyuploadtool/changelog/changelog.py

```diff
@@ -0,0 +1,106 @@
+
+import os
+
+
+# order them by priority
+# https://cheatography.com/albelop/cheat-sheets/conventional-commits/
+CONVENTIONAL_COMMIT_SPEC = {
```

Comment:
There's no reason to process any kind of "commit prefix". I don't use them, hardly anyone does as far as I can see. It adds complexity, but I don't think this is necessary.

The prefixes, if available, need to be interpreted by the user anyway. If that wasn't possible, the prefixes would be more than unnecessary bloat of the commit messages.

Reply:
> There's no reason to process any kind of "commit prefix". I don't use them, hardly anyone does as far as I can see. It adds complexity, but I don't think this is necessary.

Well, I use them, and that's one of the only blocker of not using `pyuploadtool` in my projects. I will try to show some projects using it too.

> 
> The prefixes, if available, need to be interpreted by the user anyway. If that wasn't possible, the prefixes would be more than unnecessary bloat of the commit messages.

Hmm, ok... I will still stick with this commit style.

Reply:
You can, and I have not often rejected commits because of such prefixes. The point, however, is that there is no standard or anything similar around these prefixes. _You_ may use this setup, but it's not applicable to the general public. Other people use other abbreviations, or even don't use this at all.

This "expanding" of these abbreviations is therefore IMO nothing that should be done in a tool like this one. It doesn't add any important functionality (the abbreviations can already be interpreted by the user, and git or GitHub don't expand them either). However, it adds complexity to the code base, and must be maintained. There's even a small chance that such guesswork could break other people's Git commit titles.

Reply:
I do not expand the prefixes. A trial run would reveal that. I have just replicated this, which is how a changelog'd release which follows commit spec should ideally look like:

https://github.com/srevinsaju/guiscrcpy/releases/tag/v4.7.4

Reply:
Right. I mixed this with some other project. You use it as categorization.

The criticism however stays the same. This is nothing that should be enabled by default. If you insist on this "feature", you should make it opt-in IMO.

I think you should create some `CommitSpecChangelog` which subclasses a basic `Changelog`. A factory method (e.g., in a new class) could then decide which one to instantiate. This changelog could then be inserted into the `ReleaseMetadata`. As all types of changelogs share the same interface, the release hosting providers can then, _if_ they make use of the changelog, get a formatted string, and use that to extend the release description, etc.